ext/standard/basic_functions.stub.php

```diff
@@ -3359,29 +3410,56 @@ function socket_set_timeout($stream, int $seconds, int $microseconds = 0): bool
 
 /* type.c */
 
+/**
+ * @compile-time-eval
+ */
 function gettype(mixed $value): string {}
 
+/**
+ * @compile-time-eval
+ */
 function get_debug_type(mixed $value): string {}
 
 function settype(mixed &$var, string $type): bool {}
 
+/**
+ * @compile-time-eval
+ */
 function intval(mixed $value, int $base = 10): int {}
 
+/**
+ * @compile-time-eval
+ */
 function floatval(mixed $value): float {}
 
 /** @alias floatval */
 function doubleval(mixed $value): float {}
 
+/**
+ * @compile-time-eval
+ */
 function boolval(mixed $value): bool {}
 
+/**
+ * @compile-time-eval
+ */
 function strval(mixed $value): string {}
```

Comment:
I don't know when exactly CTE applies and if there is any case where a function that is passed an object could be evaluated at compile time, so this might not apply, but for objects that implement `__toString()` the output might not be stable.

Reply:
AFAK no object/callable is evaluated on CT

Reply:
@TimWolla CTE functions are only called with arguments that are known at compile time (so excluding objects).

Reply:
Okay, good. I asked to make sure, because enums might or might not be known at compile time (those don't support `__toString()`, though), so there might also be situations for regular objects to be known statically.